.github/workflows/release.yml

```diff
@@ -38,8 +56,31 @@ jobs:
         git config user.name "github-actions[bot]"
         git config user.email "github-actions[bot]@users.noreply.github.com"
 
-    - name: Create Release
-      id: release
+    # Manual release path
+    - name: Manual Release
+      id: manual-release
+      if: github.event_name == 'workflow_dispatch'
+      run: |
+        VERSION="${{ github.event.inputs.version }}"
+
+        # Check if tag already exists
+        if git rev-parse "$VERSION" >/dev/null 2>&1; then
+          echo "Error: Tag $VERSION already exists"
+          exit 1
+        fi
+
+        # Create and push tag
+        git tag "$VERSION"
+        git push origin "$VERSION"
+
+        echo "created=true" >> $GITHUB_OUTPUT
+        echo "version=$VERSION" >> $GITHUB_OUTPUT
+        echo "tag=$VERSION" >> $GITHUB_OUTPUT
```

Comment:
I think it would be clearer to have this github workflow_dispatch only work on tags - that way the workflow itself never mutates the state of git, which feels a bit cleaner.

<img width="319" alt="image" src="https://github.com/user-attachments/assets/adaee059-4e50-456e-b5b9-bb13bbbdf471" />